src/sdk/accountspace_nb.js

```diff
@@ -215,9 +205,10 @@ class AccountSpaceNB {
     async create_access_key(params, account_sdk) {
         const action = IAM_ACTIONS.CREATE_ACCESS_KEY;
         const requesting_account = system_store.get_account_by_email(account_sdk.requesting_account.email);
-        const requested_account = validate_and_return_requested_account(params, action, requesting_account, account_sdk);
-        const account_email = params.username ? new SensitiveString(`${params.username}:${requesting_account.name.unwrap()}`) :
-                                account_sdk.requesting_account.email;
+        const requested_account = validate_and_return_requested_account(params, action, requesting_account);
+        const account_email = params.username ? account_util.get_account_name_from_username(params.username,
+                                    requesting_account._id.toString()) :
+                                requesting_account.email;
```

Comment:
You have the `requested_account`, you don't need to get the `account_email` again, right?